inky/inky.go

```diff
@@ -148,13 +193,31 @@ func New(p spi.Port, dc gpio.PinOut, reset gpio.PinOut, busy gpio.PinIn, o *Opts
 	case PHAT:
 		d.bounds = image.Rect(0, 0, 104, 212)
 		d.flipVertically = true
+	case PHAT2:
+		d.bounds = image.Rect(0, 0, 122, 250)
+		d.flipVertically = true
 	case WHAT:
 		d.bounds = image.Rect(0, 0, 400, 300)
 	}
 
 	return d, nil
 }
 
+// NewDetected tries to open a handle to an Inky pHat or wHAT automatically detected from EEPROM.
```

Comment:
Ok so eventually we could have New() be replaced to do the auto detection?

Is the I²C channel used for anything else? Is it worth forcing users to use it all the times?

Reply:
For general use it should be able to replace New() but replacing it completely would mean a user would be unable to use an Inky with faulty EEPROM, due to only being able to initialize it through detection. From reading the official python code faulty EEPROM doesn't seem to affect Inky performance, and from personal experience the python library has failed to detect one of my Inky, so this does not seem like that unlikely of a scenario.

The I²C channel is only used for reading the EEPROM, which is needed in order to detect the model. I could make NewDetected() handle opening the bus itself, but that would require using a predetermined bus and I got impression that that would be a bad approach?

Reply:
If bad EEPROM is a known failure mode, then I prefer the users to have to explicitly always call DetectOpts() (or DetectDevice() ?) then call New() themselves.

Reply:
I don't know if bad EEPROM is a known failure, this was purely anecdotal. My point was just that it could happen, so the library should probably allow manual new instances of Dev to be created. 
I am not really sure how to interpret your comment in terms of code changes, both in relation to which functions should be available and how to handle the I²C channel parameter.

Reply:
Let me clarify what I mean. My expectation is to use this flow:

```go
opts, err := inky.DetectOpts(bus)
if err != nil {
  // Use hardcoded default values.
  opts = inky.Opts{...}
}
d, err := inky.New(port, dc, reset, busy, opts)
if err != nil {
  log.Fatal(err)
}
// use d
```

So NewDetected is unnecessary.

Reply:
Yeah okay, sounds good I'll get on it. I think that is what I was trying to express in the conversation below this one